packages/cli/src/util/build/monorepo.ts

```diff
@@ -1,40 +1,28 @@
-import fs from 'fs-extra';
-import { join, relative, basename } from 'path';
+import { relative, basename } from 'path';
 import {
-  detectFramework,
-  monorepoManagers,
   LocalFileSystemDetector,
-  packageManagers,
+  getMonorepoDefaultSettings,
+  MissingBuildPipeline,
+  MissingBuildTarget,
 } from '@vercel/fs-detectors';
 import { ProjectLinkAndSettings } from '../projects/project-settings';
 import { Output } from '../output';
 import title from 'title';
-import JSON5 from 'json5';
+import { PartialProjectSettings } from '../input/edit-project-settings';
 
 export async function setMonorepoDefaultSettings(
   cwd: string,
   workPath: string,
-  projectSettings: ProjectLinkAndSettings['settings'],
+  projectSettings: ProjectLinkAndSettings['settings'] & PartialProjectSettings,
   output: Output
 ) {
   const localFileSystem = new LocalFileSystemDetector(cwd);
 
-  const [monorepoManager, packageManager] = await Promise.all([
-    detectFramework({
-      fs: localFileSystem,
-      frameworkList: monorepoManagers,
-    }),
-    detectFramework({
-      fs: localFileSystem,
-      frameworkList: packageManagers,
-    }),
-  ]);
-
   const projectName = basename(workPath);
   const relativeToRoot = relative(workPath, cwd);
 
   const setCommand = (
-    command: 'buildCommand' | 'installCommand',
+    command: 'buildCommand' | 'installCommand' | 'commandForIgnoringBuildStep',
```

Comment:
**suggestion (non-blocking):** this PR is described as moving logic, but it also appears to add logic around `commandForIgnoringBuildStep`. In the future, can you either use multiple PRs or mention changes like this in the description?

Reply:
yep my bad. @chloetedder and i were on the same page about this but I forgot to update things appropriately.